src/functions/functionalPolicyFactory.ts

```diff
@@ -0,0 +1,55 @@
+import { APIGatewayAuthorizerResult, Statement } from "aws-lambda";
+import newPolicyDocument from "./newPolicyDocument";
+import { ILogEvent } from "../models/ILogEvent";
+import { HttpVerb } from "../services/http-verbs";
+import StatementBuilder from "../services/StatementBuilder";
+
+type NonEmptyArray<T> = [T, ...T[]];
+
+interface IApiAccess {
+  verb: HttpVerb;
+  path: string;
+}
+
+const functionConfig: { [key: string]: NonEmptyArray<IApiAccess> } = {
+  "TechRecord.Amend": [
+    {
+      verb: "POST",
+      path: "vehicles/*",
+    },
+    {
+      verb: "PUT",
+      path: "vehicles/*",
+    },
+  ],
+  "TechRecord.View": [
+    {
+      verb: "GET",
+      path: "vehicles/*",
+    },
+  ],
+};
+
+function toStatement(access: IApiAccess): Statement {
+  return new StatementBuilder().setEffect("Allow").setHttpVerb(access.verb).setResource(access.path).build();
+}
+
+export function generatePolicy(jwt: any, logEvent: ILogEvent): APIGatewayAuthorizerResult | undefined {
```

Comment:
Would be nice to have some unit tests for this new function

Reply:
Realised you've somewhat tested this function through the new `authoriser` tests but would be good to see that the ARN policies look as expected and not just checking that the length is equal to 3 in this test
`it("should return multiple statements when multiple roles are valid", async () => {`

Reply:
Added test to ensure policy format is being produced correctly